camunda/camunda
# feat: update ilm and ism polices for existent indexes

## Description

For Elastic Search - The implementation resumes in only apply updates for Existent indexes, (no need to apply index template setup, as it uses phases, to validate that)

References from ISM implementation 
https://opensearch.org/docs/latest/im-plugin/ism/api (Note ISM does not support Patch opertions), So I am using PUT preserving the old configurations, and only including the new one.
- The Java Client API does not support the ISM, also the policy of open search is not associated to the phasis, but to the index_template settings §

**To Test (Development tests)**

-  https://docs.camunda.io/docs/self-managed/tasklist-deployment/data-retention/#snippet-from-applicationyml  Check this document to understand how to activate ilm 
- Setup archive with small time
- Start Tasklist Opensearch/ElasticSearch
- Turn on/off ILM flag 

On elastic search you can validate looking the indexes templates settings over here:
localhost:9200/tasklist-task*/_settings  (Check the policy exists or does not, according to the setting)

For open search: 
localhost:9205/tasklist-task*/_settings

Note: The policy is only applied to the indexes with date suffix.

I could not find a good way to work on integration tests for this ticket, the IT does not exists neither for ILM or ISM, I purpose the follow up ticket 
- We dont have a integration test neither for ISM or ILM, For the next spring cleaning I would suggest a ticket to implement this. Maybe a nice first ticket for onboarding. 

## Related issues

closes https://github.com/camunda/tasklist/issues/4771
